### PR TITLE
Feature/case insensitive header validation

### PR DESCRIPTION
**Pull request by maciakm**
I replaced all uses of ->getHeaders()[$headerName] that are case sensitive with corresponding methods from MessageInterface (that are case insensitive)
plus some minor refactoring

### CODE REVIEW

File: src/PSR7/Validators/Headers.php [hunk 1]
```diff
@@ -20,22 +19,14 @@ class Headers
...
-                // TODO, maybe this can be enabled later and controlled by custom options
-                // throw new \RuntimeException($header, 200);
-                continue;
-            }
-
-            $validator = new SchemaValidator($this->detectValidationStrategy($message));
-            foreach ($headerValues as $headerValue) {
-                $validator->validate($headerValue, $headerSpecs[$header]->schema);
-            }
+        if (count($headerSpecs) == 0) {
```
Comment (scaytrase) [inline src/PSR7/Validators/Headers.php]: Here is no need in explicit check, since with empty count there will be just an empty foreach.

File: src/PSR7/Validators/Headers.php [hunk 2]
```diff
@@ -20,22 +19,14 @@ class Headers
      */
     public function validate(MessageInterface $message, array $headerSpecs) : void
     {
-        $messageHeaders = $message->getHeaders();
+        // By default this will not report unexpected headers (soft validation)
```
Comment (scaytrase) [inline src/PSR7/Validators/Headers.php]: Leaving this commit with removed header iteration does no sence, since implementing todo will mean revert this change back

**Comment by lezhnev74** [comment]
Ok, looks better now. Merging.
